plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -89,29 +113,45 @@ private static boolean hasUserOptedIn() {
    * opting out. (Therefore, this method essentially returns true if the opt-in dialog was ever
    * presented before.)
    */
-  private static boolean hasUserRegisteredOptInStatus() {
-    return getBooleanPreference(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, false);
+  private boolean hasUserRegisteredOptInStatus() {
+    return preferences.getBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, false);
   }
 
-  static void registerOptInStatus(boolean optedIn) {
-    IEclipsePreferences prefs = ConfigurationScope.INSTANCE.getNode(PREFERENCES_PLUGIN_ID);
-    if (prefs != null) {
-      prefs.putBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN, optedIn);
-      prefs.putBoolean(CloudToolsPreferencePage.ANALYTICS_OPT_IN_REGISTERED, true);
-      flushPreferences(prefs);
-    }
+  void registerOptInStatus(boolean optedIn) {
```

Comment:
I agree this can be confusing. OPT_IN and OPT_IN_REGISTERED are certainly related. However, they are also independent in some sense. I added some more comments for clarification.

Details:
It would have been ideal if I had a three-valued preference setting: OPTED_IN, OPTED_OUT, and UNDEF. However, such setting will require implementing a custom UI field editor for the preference page, so I took a simple approach to use two Boolean settings (so that I can just use a single `BooleanFieldEditor` in the UI).